sdk/storage/azure-storage-file-datalake/out/azure/storage/filedatalake/_data_lake_directory_client.pyi

```diff
@@ -0,0 +1,23 @@
+# Stubs for azure.storage.filedatalake._data_lake_directory_client (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from ._data_lake_file_client import DataLakeFileClient
+from ._models import DirectoryProperties
+from ._path_client import PathClient
+from ._shared.base_client import parse_connection_str
+from typing import Any, Optional
+
+class DataLakeDirectoryClient(PathClient):
+    def __init__(self, account_url: str, file_system_name: str, directory_name: str, credential: Optional[Any]=..., **kwargs: Any) -> None: ...
+    @classmethod
+    def from_connection_string(cls: Any, conn_str: str, file_system_name: str, directory_name: str, credential: Optional[Any]=..., **kwargs: Any) -> DataLakeDirectoryClient: ...
+    def create_directory(self, metadata: Optional[Dict[str, str]]=..., **kwargs: Any) -> Dict[str, Union[str, datetime]]: ...
+    def delete_directory(self, **kwargs: Any) -> None: ...
+    def get_directory_properties(self, **kwargs: Any) -> DirectoryProperties: ...
+    def rename_directory(self, rename_destination: Any, **kwargs: Any): ...
```

Comment:
Should `rename_directory` be type `str`? or do we support other inputs here?

Reply:
Also, in `FileClient.rename_file` the parameter is just called `new_name`.

Reply:
Good catch! I forgot to add type annotation here! This will be addressed in an existing pr, and rename_destination is changed to new_name there.